app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
I won't hold the PR up, but I am not a big fan of this pattern as it feels like overloading. A singleton that creates instances of itself? A RepositoryTypeManager that manages instances of RepositoryTypeManager objects? IMO, it would be clearer to have a RepositoryTypeManager singleton or object that has a collection of instances of RepositoryType objects that it is managing.